### PR TITLE
Gra 1264 better peer mgmt

### PR DESCRIPTION
**Pull request by 0xdcarns**
## Describe your changes
Removed some useless code. Fixed handling deleting peers better. Pairs with netclient story/GRA-1264-better-peer-mgmt

## Provide testing steps
- [x] Install netclient on a few hosts
- [x] Join multiple networks with 3+ peers in each
- [x] Upon joining ensure WG connections were formed
- [x] Delete node from one and ensure allowed ips are updated
- [x] Delete the same completely and ensure peer is removed quickly while not affecting other connections

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [x] If it is a bugfix, my code is <= 200 lines.
- [x] My functions are <= 80 lines.
- [x] I have had my code reviewed by a peer.
- [x] My unit tests pass locally.
- [x] Netmaker is awesome.

### CODE REVIEW

File: logic/peers.go [hunk 1]
```diff
@@ -514,9 +352,32 @@ func GetPeerUpdateForHost(network string, host *models.Host) (models.HostPeerUpd
 		}
 	}
 
+	// run through delete nodes
+	if len(deletedNodes) > 0 {
```
Comment (mattkasun) [inline logic/peers.go:356]: unnecessary check as for loop will be skipped if deletedNodes is empty.  The compiler will probably optimize out the check so not a big deal

**Review by mattkasun — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
all check marks pass
nodes maintained connectivity throughout.
pass

File: logic/peers.go [hunk 2]
```diff
@@ -328,10 +159,14 @@ func GetPeerUpdateForHost(network string, host *models.Host) (models.HostPeerUpd
...
 		for _, peer := range currentPeers {
+			peer := peer
 			if peer.ID == node.ID {
 				logger.Log(2, "peer update, skipping self")
 				//skip yourself
-
+				continue
+			}
+			if peer.Action == models.NODE_DELETE || peer.PendingDelete {
+				deletedNodes = append(deletedNodes, peer) // track deleted node for peer update
```
Comment (abhishek9686) [inline logic/peers.go:169]: wouldn't this approach only add zombie nodes to deleted nodes list, and peers which are deleted already would never be removed on client

Comment (0xdcarns) [reply logic/peers.go:169]: the zombie check timeout was extended to discount for this

**Pull request merged**